Content.Server/Storage/Components/EntityStorageComponent.cs

```diff
@@ -165,14 +165,21 @@ protected override void Initialize()
 
         public virtual void Activate(ActivateEventArgs eventArgs)
         {
+            // HACK until EntityStorageComponent gets refactored to the new ECS system
+            if (Owner.TryGetComponent<LockComponent>(out var @lock) && @lock.Locked)
+            {
+                // Do nothing, LockSystem is responsible for handling this case
+                return;
+            }
```

Comment:
This is a good candidate for using the eventbus instead and if the event is cancelled then return instead.

Reply:
would it be ok, if I did this change as part of a EntityStorageComponent refactor?